### ISSUE TITLE
consensusString and consensusMatrix error for legth-0 input

### ISSUE THREAD

**Issue by jan-glx** [Bioconductor/Biostrings#101]
Currently, `consensusString` fails for length-0 inputs:
``` r
Biostrings::consensusString(character(0))
#> Error in h(simpleError(msg, call)): error in evaluating the argument 'x' in selecting a method for function 'consensusString': 'x' has no element and 'width' is NULL
```

<sup>Created on 2023-07-29 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

I believe it should return `character(0)` instead and at *most* warn.

*Maybe*, replacing [`error("'x' has no element and 'width' is NULL");`](https://github.com/Bioconductor/Biostrings/blob/35899c8dc44efae58038d81248c6122323072b9a/src/letter_frequency.c#L802C19-L802C19)  with `return allocMatrix(INTSXP, 0, 0);` is all that's needed?

**Comment by ahl27** [comment]
Thanks for the callout, and apologies for the slow turnaround--this is fixed in #111.

The return values in the new version should be consistent with inputs of length > 0:
- `consensusMatrix(..., as.prob=FALSE)` returns an integer matrix
- `consensusMatrix(..., as.prob=TRUE)` returns a numeric matrix
- `consensusString(x)` returns `character(0L)` if the input is length 0
- `consensusString(DNAStringSet())` has the correct column names (similarly for all other `XStringSet` objects, and for `baseOnly=TRUE`)

I'll merge this sometime this month once I've double checked that it's not causing any other weird bugs.

**Comment by ahl27** [comment]
just resolved in devel this week.
